rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -882,13 +1076,46 @@ void RSocketStateMachine::writeNewStream(
         streamId, RequestOriginator::LOCAL, streamToken, streamType);
   }
 
+  // for simplicity, require that sent buffers don't consist of chains
+  if ((bufChainedLength(payload.metadata) + bufChainedLength(payload.data)) >
+      GENEROUS_MAX_FRAME_SIZE) {
+    detail::writeFragmented(
+        [&](Payload p) {
```

Comment:
this is only for some individuals with strong stomach. ;) Why would you not pass streamType into the writeFragmented method and inline the lambda inside the function too?

Can we put all of this logic to a separate class so that RSStateMachine is no blowing up in size and responsibilities?